test/integration/logins-api-test.js

```diff
@@ -235,6 +249,45 @@ describe("logins API", () => {
     expect(logins[0].timesUsed).to.equal(2);
   });
 
+  describe("browser.experiments.logins.{set,get}LoginSavingEnabled", () => {
+    const commonTest = async (button, expected) => {
+      await loadTestPage();
+      await clickButton(button ? "enable-saving" : "disable-saving");
+      const results = await driver.wait(until.elementLocated(
+        By.id("saving-results")
+      ), 1000);
+      await driver.wait(
+        until.elementTextContains(
+          results,
+          expected ? "true" : "false"
+        ),
+        5000
+      );
+    };
+    describe("with pref signon.rememberSignons = true", () => {
+      before(async () => {
+        await enableRememberSignons();
+      });
+      after(async () => {
+        await disableRememberSignons();
+      });
+      it("disables login saving for origin when given false", async () => {
+        await commonTest(false, false);
+      });
+      it("enables login saving for origin when given true", async () => {
+        await commonTest(true, true);
+      });
+    });
+    describe("with pref signon.rememberSignons = false", () => {
+      before(async () => {
+        await disableRememberSignons();
+      });
+      it("does not enable login saving for origin when given true", async () => {
+        await commonTest(true, false);
```

Comment:
Mild corner case here: If the user has turned off saving logins globally in options, `getLoginSavingEnabled` always returns false no matter what the setting might be.

Reply:
Good catch! I'm not sure which, if any, parts of the logins API will work if logins are globally disabled. I think this needs investigation; could you file a bug?

Reply:
From my own experiences, the CRUD portion of the upstream API works just fine regardless of what the global pref and this specific callout are set to.